tests/unit/test_validate_checkpoint_operator.py

```diff
@@ -66,3 +70,143 @@ def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+        # arrange
+        context_type: Literal["ephemeral"] = "ephemeral"
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+            return Mock()
+
+        mock_gx = Mock()
+        mocker.patch.dict("sys.modules", {"great_expectations": mock_gx})
+        validate_checkpoint = GXValidateCheckpointOperator(
+            task_id="validate_checkpoint",
+            configure_checkpoint=configure_checkpoint,
+            context_type=context_type,
+        )
+
+        # act
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_called_once_with(mode=context_type)
+
+    def test_context_type_cloud(self, mocker: MockerFixture) -> None:
+        """Expect that param context_type creates a CloudDataContext."""
+        # arrange
+        context_type: Literal["cloud"] = "cloud"
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+            return Mock()
+
+        mock_gx = Mock()
+        mocker.patch.dict("sys.modules", {"great_expectations": mock_gx})
+        validate_checkpoint = GXValidateCheckpointOperator(
+            task_id="validate_checkpoint_success",
+            configure_checkpoint=configure_checkpoint,
+            context_type=context_type,
+        )
+
+        # act
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_called_once_with(mode=context_type)
+
+    def test_context_type_filesystem(self, mocker: MockerFixture) -> None:
+        """Expect that param context_type defers creation of data context to user."""
+        # arrange
+        context_type: Literal["file"] = "file"
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+            return Mock()
+
+        def configure_file_data_context() -> FileDataContext:
+            return Mock()
+
+        mock_gx = Mock()
+        mocker.patch.dict("sys.modules", {"great_expectations": mock_gx})
+        validate_checkpoint = GXValidateCheckpointOperator(
+            task_id="validate_checkpoint_success",
+            configure_checkpoint=configure_checkpoint,
+            context_type=context_type,
+            configure_file_data_context=configure_file_data_context,
+        )
+
+        # act
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_not_called()
+
+    def test_user_configured_context_is_passed_to_configure_checkpoint(self) -> None:
+        """Expect that if a user configures a file context, it gets passed to the configure_checkpoint function."""
+        # arrange
+        context_type: Literal["file"] = "file"
+        mock_file_data_context = Mock()
+
+        def configure_file_data_context() -> FileDataContext:
+            return mock_file_data_context
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
```

Comment:
Probably easier to just pass a mock for this, and assert that the mock was passed the right context, but either way works.

Reply:
yep, [479dd72](https://github.com/great-expectations/gx-airflow-provider/pull/8/commits/479dd726673d04b771697a2c339ed7d08cae40ae), thanks.